feature-libs/customer-ticketing/components/list/customer-ticketing-list/customer-ticketing-list.component.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component } from '@angular/core';
+import { RoutingService, TranslationService } from '@spartacus/core';
+import {
+  CustomerTicketingFacade,
+  STATUS,
+  TEXT_COLOR_CLASS,
+  TicketList,
+} from '@spartacus/customer-ticketing/root';
+import { ICON_TYPE } from '@spartacus/storefront';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-customer-ticketing-list',
+  templateUrl: './customer-ticketing-list.component.html',
+})
+export class CustomerTicketingListComponent {
+  constructor(
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected routing: RoutingService,
+    protected translation: TranslationService
+  ) {}
+  PAGE_SIZE = 5;
+  sortType: string = 'byId';
+  iconTypes = ICON_TYPE;
+  customerTicketsFlag: boolean = true;
+  tickets$: Observable<TicketList | undefined> =
+    this.customerTicketingFacade.getTickets(this.PAGE_SIZE);
+
+  goToTicketDetail(ticketId: string): void {
+    this.routing.go({
+      cxRoute: 'supportTicketDetails',
+      params: { ticketCode: ticketId },
+    });
+  }
+
+  getSortLabels(): Observable<{ byTicketId: string; byDate: string }> {
+    return combineLatest([
+      this.translation.translate('customerTicketing.ticketId'),
+      this.translation.translate('customerTicketing.changedOn'),
+    ]).pipe(
+      map(([textByTicketId, textByDate]) => {
+        return {
+          byTicketId: textByTicketId,
+          byDate: textByDate,
+        };
+      })
+    );
+  }
+
+  changeSortCode(sortCode: string): void {
+    this.sortType = sortCode;
+    this.pageChange(0);
+  }
+
+  pageChange(page: number): void {
+    const ticketListParams = this.createTicketListEvent(this.sortType, page);
+    this.fetchTicketList(ticketListParams);
+  }
+
+  createTicketListEvent(
+    sortCode: string,
+    currentPage: number
+  ): { currentPage: number; sortCode: string } {
+    return {
+      currentPage: currentPage,
+      sortCode: sortCode,
+    };
+  }
+
+  private fetchTicketList(ticketListParams: {
+    sortCode: string;
+    currentPage: number;
+  }): void {
+    this.tickets$ = this.customerTicketingFacade.getTickets(
```

Comment:
Going back to this. I have discussed with a peer, and I recalled that this was not supposed to be handled this way through a query. What was supposed to be done specifically for the get ticket list request was to call the `connector` directly instead of wrapping it in a query. https://sap-my.sharepoint.com/:w:/p/parth_lakhani/EX96033xTSdMsG_zhI41Hh4Bn43TOKuOb1gpKNGmss_ABQ?e=k11ABj @adolfobee @Parthlakhani

Reply:
after several discussions with Parth and a peer, we'll give it for now